Doc/library/sqlite3.rst

```diff
@@ -302,10 +302,10 @@ Module functions
    :type isolation_level: str | None
 
    :param bool check_same_thread:
-       If ``True`` (default), only the creating thread may use the connection.
-       If ``False``, the connection may be shared across multiple threads;
-       if so, write operations should be serialized by the user to avoid data
-       corruption.
+       If ``True`` (default), an exception will be raised if the database connection
+       is used by a thread other than the one that created it.
```

Comment:
Can we be more precise about what "used" means here? Simply if the top-level object is accessed in any manner? Calling specific read and/or write methods?

Reply:
Well, the SQLite docs use similar wording and is just as vague (https://sqlite.org/threadsafe.html).

_I_ assume this means usage at the data structure level, so for example for the _SQLite multi-threaded_ mode, I assume all C APIs that work with `sqlite3 *` pointers are _not_ thread safe, but all C APIs that work with `sqlite3_stmt *` pointers are thread safe. This is hard to translate to Python APIs, though, as there is no one-to-one mapping of sqlite3 Python APIs and SQLite C APIs.

For example, most of the Python APIs of `sqlite3.Cursor` map to SQLite C APIs that deal with `sqlite3_stmt *`; this is fine... until an exception is raised! All exceptions raised by the sqlite3 module use C APIs that access the current connection. Thread safe? I don't know; perhaps you'll end up with the wrong error message? Or perhaps you'll end up with no error message at all, in case the sqlite3 module will return `NULL` from a Py\_ C API _without_ an exception being set, which is a fatal error; you'll get a seg fault, IIRC.

If the underlying SQLite module is using **serialized** mode, you can of course safely set `check_same_thread` to `False`.